crypto/src/interface.rs

```diff
@@ -0,0 +1,26 @@
+use core::result::Result;
+
+use serde::{de::DeserializeOwned, ser::Serialize};
+
+pub trait Contribution: Serialize + DeserializeOwned {
+    type Receipt: Serialize;
+    fn get_receipt(&self) -> Self::Receipt;
```

Comment:
In a follow up PR we'll want to extend this with methods useful for client side.